specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2022-04-01/componentPolicyStates.json

```diff
@@ -0,0 +1,843 @@
+          "type": "string",
+          "readOnly": true
+        },
+        "expressionKind": {
+          "description": "The kind of expression that was evaluated.",
+          "type": "string",
+          "readOnly": true
+        },
+        "path": {
+          "description": "Property path if the expression is a field or an alias.",
+          "type": "string",
+          "readOnly": true
+        },
+        "expressionValue": {
+          "description": "Value of the expression.",
+          "type": "object",
+          "readOnly": true
+        },
+        "targetValue": {
+          "description": "Target value to be compared with the expression value.",
+          "type": "object",
+          "readOnly": true
+        },
+        "operator": {
+          "description": "Operator to compare the expression value and the target value.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    }
+  },
+  "parameters": {
+    "componentPolicyStatesResource": {
+      "name": "componentPolicyStatesResource",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "enum": [
+        "latest"
+      ],
+      "x-ms-enum": {
+        "name": "ComponentPolicyStatesResource",
+        "modelAsString": true
+      },
+      "description": "The virtual resource under ComponentPolicyStates resource type. In a given time range, 'latest' represents the latest component policy state(s).",
+      "x-ms-parameter-location": "method"
+    },
+    "subscriptionId": {
+      "name": "subscriptionId",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "Microsoft Azure subscription ID.",
+      "x-ms-parameter-location": "method"
+    },
+    "resourceGroupName": {
+      "name": "resourceGroupName",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "Resource group name.",
+      "x-ms-parameter-location": "method"
+    },
+    "resourceId": {
+      "name": "resourceId",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "Resource ID.",
+      "x-ms-parameter-location": "method",
+      "x-ms-skip-url-encoding": true
+    },
+    "authorizationNamespace": {
+      "name": "authorizationNamespace",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "enum": [
+        "Microsoft.Authorization"
+      ],
+      "x-ms-enum": {
+        "name": "AuthorizationNamespaceType",
+        "modelAsString": false
+      },
+      "description": "The namespace for Microsoft Authorization resource provider; only \"Microsoft.Authorization\" is allowed.",
+      "x-ms-parameter-location": "method"
+    },
+    "policyDefinitionName": {
+      "name": "policyDefinitionName",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "Policy definition name.",
+      "pattern": "^[a-zA-Z0-9][a-zA-Z0-9_.-]*$",
```

Comment:
@pilor What regex pattern should we use? Just allow everything? I don't see an equivalent pattern in Auth RPs validation code or anything

Reply:
the maxLength 64 is fine.  Character-wise Auth RP does not have its own restrictions beyond what ARM enforces: 

Something like `"^[^<>%&:\\\\?/#]*$"` is probably correct but do some sanity checks on that.

Reply:
Thanks Chris also validated CLI today for definition creates returns the following for invalid characters:  `The resource name has invalid characters '%, &, \, ?, /, <, >, :' or control characters. `